### ISSUE TITLE
Modifier key combinations don't send key up events to framework. 

### ISSUE THREAD

**Issue by gspencergoog** [flutter/flutter#44135]
Currently, the macOS shell seems to receive the key-up event for command sequences like "⌘-C" and "Shift-A" without producing a key up event in the framework for the letters, which is problematic on the framework side when someone wants to type, much less re-map it to something else, and it causes the framework to think that the non-modifier keys never went up, so they stick until someone presses the key again (e.g. if I press "⌘-C", the framework thinks that the "C" key is pressed until it gets pressed again).

**Comment by gspencergoog** [comment]
cc @franciscojma86

**Comment by gspencergoog** [comment]
You'll likely need to apply #44130 before you can see what I mean (until it lands).

**Comment by shyndman** [comment]
This might be worth surfacing again. I'm continuing to run this code with no issues.
https://github.com/flutter/flutter/issues/38076#issuecomment-522784379

**Comment by gspencergoog** [comment]
Thanks @shyndman, we'll definitely look at that method and evaluate it.

**Comment by franciscojma86** [comment]
Taking a look at this, it seems to be an OS limitation. It seems to only happen with command. While @shyndman suggestion does work, I worry that it might have an unexpected impact given that we would be overriding NSApplication. Another downside I see of that approach, is that it would have to be overriden from the runner app. 

@gspencergoog maybe this is something we can workaround on the framework side, to "free" the pressed key when command is released? FWIW if you release cmd before the key, `keyUp` is called

**Comment by franciscojma86** [comment]
My suggestion wouldn't actually work. Will keep looking into how we can get the raw event.

**Comment by shyndman** [comment]
I have something to consider.

Both Windows and macOS shortcuts work by evaluating a single keyboard key + a set of zero or more modifiers. The advantage of this approach is that a single keydown event describes whether or not to trigger the shortcut's action — no need to store comprehensive keyboard state. 

That said, I appreciate that in the current approach you can ask whether a key is down at any point. Useful. You can do this easily in Windows, too, but not in macOS without using lower-level APIs.

**Comment by franciscojma86** [comment]
@shyndman the behavior you suggest was implemented recently (using the modifies within the NSEvent of the pressed key) so that's indeed a good observation 😄. However, that doesn't solve the problem at hand since we don't even get the NSEvent from the key being released. I just posted a PR for review, using an approach similar to the one you suggested, but using a listener. Thanks for your suggestion!
